Contrib/ChiralPairs/ChiralDescriptors.py

```diff
@@ -0,0 +1,300 @@
+                                                 sharedNeighbors=numShared, numRotBonds=numRotBs, numAroBonds=aroBonds)
+    # if we have less then 4 substituents the missing ones need to be an hydrogen atoms
+    if len(sizeDict) < 4:
+        for i in range(4-len(sizeDict)):
+            sizeDict['H'+str(i)]=substituentDescriptor(size=0,relSize=0,numNO=0,relNumNO=0,relNumNO_2=0,
+                                                       pathLength=0,relPathLength=0,relPathLength_2=0,sharedNeighbors=0, numRotBonds=0,
+                                                       numAroBonds=0)
+    return sizeDict
+
+
+# Visualization of the substituents
+def visualizeSubstituentsGrid(mol, aIdx, molSize=(300,150), kekulize=True,):
+    dists = Chem.GetDistanceMatrix(mol)
+    idxChiral = Chem.FindMolChiralCenters(mol)[0][0]
+    subDict, sn_dict, maxShell_dict = determineAtomSubstituents(aIdx, mol, dists, False)
+    
+    colors = sns.husl_palette(len(subDict), s=.6)    
+    mc = rdMolDraw2D.PrepareMolForDrawing(mol, kekulize=kekulize)
+    count=0
+    svgs=[]
+    labels=[]
+    for sidx, sub in sorted(subDict.items(), key= lambda x: _getSizeOfSubstituents(x[1], sn_dict)):
+        color = tuple(colors[count])        
+        count+=1
+        atColors={}
+        for atom in sub:
+            atColors[atom]=color
+        
+        bonds = getBondsSubstituent(mol, set(sub)) 
+        bnColors={}
+        for b in bonds:
+            bnColors[b]=color
+        
+        drawer = rdMolDraw2D.MolDraw2DSVG(molSize[0],molSize[1])
+        drawer.DrawMolecule(mc,highlightAtoms=atColors.keys(),
+                            highlightAtomColors=atColors,highlightBonds=bonds,highlightBondColors=bnColors)
+        drawer.FinishDrawing()
+        svg = drawer.GetDrawingText()
+        svgs.append(svg.replace('svg:',''))
+        labels.append("Substituent "+str(count)+" (#atoms: "+str(len(sub))+", size normed: "+str(_getSizeOfSubstituents(sub, sn_dict))+")")
+    return _svgsToGrid(svgs, labels, svgsPerRow=len(svgs),molSize=molSize,fontSize=12)    
+    
+def visualizeChiralSubstituentsGrid(mol):
+    idxChiral = Chem.FindMolChiralCenters(mol)[0][0]
+    return visualizeSubstituentsGrid(mol, idxChiral)
+
+# Chiral moment descriptor    
+from numpy import linalg as LA
+def calcSP3CarbonSubstituentMoment(subSizes):
+
+    if len(subSizes) != 4:
+        raise ValueError('Function "calcSP3CarbonSubstituentMoment" expects an array of size 4 as parameter')
+        
+    # tetrahedron unit vectors
+    x1=np.array([1,1,1])
+    x2=np.array([-1,1,-1])
+    x3=np.array([1,-1,-1])
+    x4=np.array([-1,-1,1])
+
+    substituentMoment= LA.norm((subSizes[0]*x1)+(subSizes[1]*x2)+(subSizes[2]*x3)+(subSizes[3]*x4))
+    return substituentMoment
+
+def generateChiralDescriptors(mol):
+    desc={}
+    dists = Chem.GetDistanceMatrix(mol)
+    idxChiral = Chem.FindMolChiralCenters(mol)[0][0]
```

Comment:
This limits the function to pick the first chiral center, doesn't it? Could you make it more general to give the user the option to pick one? You could even add a method that returns the descriptor for each of the chiral centers.